lib/swoosh/adapters/postmark.ex

```diff
@@ -67,26 +75,53 @@ defmodule Swoosh.Adapters.Postmark do
   defp prepare_cc(body, %Email{cc: []}), do: body
   defp prepare_cc(body, %Email{cc: cc}), do: Map.put(body, "Cc", prepare_recipients(cc))
 
-  defp prepare_bcc(body, %Email{bcc: []}), do: body
+  defp prepare_bcc(body, %Email{bcc: []}),  do: body
   defp prepare_bcc(body, %Email{bcc: bcc}), do: Map.put(body, "Bcc", prepare_recipients(bcc))
 
-  defp prepare_reply_to(body, %Email{reply_to: nil}), do: body
+  defp prepare_reply_to(body, %Email{reply_to: nil}),              do: body
   defp prepare_reply_to(body, %Email{reply_to: {_name, address}}), do: Map.put(body, "ReplyTo", address)
 
   defp prepare_recipients(recipients) do
     recipients
-    |> Enum.map(&prepare_recipient(&1))
+    |> Enum.map(&prepare_recipient/1)
     |> Enum.join(",")
   end
 
-  defp prepare_recipient({"", address}), do: address
+  defp prepare_recipient({"",   address}), do: address
   defp prepare_recipient({name, address}), do: "\"#{name}\" <#{address}>"
 
+  defp prepare_subject(body, %Email{subject: ""}),      do: body
   defp prepare_subject(body, %Email{subject: subject}), do: Map.put(body, "Subject", subject)
 
-  defp prepare_text(body, %{text_body: nil}), do: body
-  defp prepare_text(body, %{text_body: text_body}), do: Map.put(body, "TextBody", text_body)
+  defp prepare_text(body, %Email{text_body: nil}),       do: body
+  defp prepare_text(body, %Email{text_body: text_body}), do: Map.put(body, "TextBody", text_body)
+
+  defp prepare_html(body, %Email{html_body: nil}),       do: body
+  defp prepare_html(body, %Email{html_body: html_body}), do: Map.put(body, "HtmlBody", html_body)
+
+  # example custom vars
+  #
+  # %{
+  #   "template_id"    => 123,
+  #   "template_model" => %{"name": 1, "company": 2}
+  # }
+  defp prepare_custom_vars(body, %Email{provider_options: provider_options}),
+    do: Enum.reduce(provider_options, body, &put_in_body/2)
```

Comment:
Feels dangerous to reduce the whole `provider_options` into the body of the request.

Feels like this would be better handled by a specific `prepare_template/2` function that would match exactly on these two keys and insert them into the body.